compiler/rustc_middle/src/ty/context.rs

```diff
@@ -418,6 +418,12 @@ pub struct TypeckResults<'tcx> {
     /// Stores the type, expression, span and optional scope span of all types
     /// that are live across the yield of this generator (if a generator).
     pub generator_interior_types: Vec<GeneratorInteriorTypeCause<'tcx>>,
+
+    /// We sometimes treat byte string literals (which are of type`&[u8; N]`)
+    /// as `&[u8]`, depending on the pattern  in which they are used.
+    /// This hashset records all instances where we behave
+    /// like this to allow other code to reliably process this.
```

Comment:
"other code", could we explicitly mention `const_to_pat`?

Reply:
:laughing: I had this before, but worried that this documentation would then just regress, but I guess I also don't see what other could could possibly care for this information, so yea, I'll add it again